plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/decoders/ArrayDecoder.java

```diff
@@ -42,11 +42,16 @@ public ArrayDecoder(Type type)
     public void decode(Supplier<Object> getter, BlockBuilder output)
     {
         List<?> value = (List<?>) getter.get();
-        BlockBuilder elementBlockBuilder = type.getElementType().createBlockBuilder(null, 1);
-        for (int i = 0; i < value.size(); i++) {
-            int index = i;
-            elementDecoder.decode(() -> value.get(index), elementBlockBuilder);
+        if (value == null) {
+            output.appendNull();
+        }
+        else {
```

Comment:
looks like unrelated to aggregations, is it?

Seems like you're fixing reading of array type when the array itself is null?

Reply:
The only time it could be null is when it's set during a single row aggregation (no grouping columns) over an empty set, depending on the aggregate function. Counting functions return 0 for empty set and everything else returns null. Pinot does not return null values, the connector sets them to null in this case.